vagrant-kubernetes-istio/README.md

```diff
@@ -1,19 +1,103 @@
-This test environment is tested on Ubuntu 16.04 (host), virtualbox 5.2.8, docker 17.09.0-ce, kubernetes 1.10.
- - Build Istio docker images on host machine.
- - Set up private docker registry in Kubernetes on VM.
- - Push images from host to VM. The images are stored in private docker registry
- - Run Istio e2e tests command on Host. The test environment is deployed in Kubernetes on VM, and test result is returned to host.
+# Prereqs:
+1. Set up Istio Dev envrionment using https://github.com/istio/istio/wiki/Dev-Guide.
 
+1. Install
+  * [virtual box](https://www.virtualbox.org/wiki/Downloads) - Verify `virtualbox` command opens up a virtual box window
+  * [docker](https://docs.docker.com/) - Verify `docker version` returns version >= 18.03.0-ce
+  * [vagrant](https://www.vagrantup.com/downloads.html) - Verify `vagrant -v` returns version >= Vagrant 2.0.3
+  * [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl) - Verify `kubectl version` returns both server and client versions
+  * [curl](https://curl.haxx.se/)
 
-# [RunTestOnVm](https://github.com/istio/istio/tree/vagrant-kubernetes-istio/vagrant-kubernetes-istio/RunTestOnVm "RunTestOnVm")
-This test environment is tested on Ubuntu 16.04 (host), virtualbox 5.2.8, docker 17.09.0-ce, kubernetes 1.10.
- - Build Istio docker images on host machine.
- - Set up private docker registry in Kubernetes on VM.
- - Push images from host to VM. The images are stored in private docker registry
- - Run Istio e2e tests command on VM. The test result is returned to VM.
+You can run the following OS specific scripts to install all pre-requisites, or use them as a reference to install them manually. .
 
+```bash
+// assumes [homebrew](https://brew.sh) exists for Mac
+sh prereqs.sh
+```
+
+# Steps
+## 1. Set up Vagrant Environment
+```bash
+sh vm_setup.sh
+```
+
+## 2. Set up Docker daemon and kubectl on Host
+```bash
+sh host_setup.sh
+```
+If you are on macOS, you need to setup docker daemon using UI additionally.
+Click on the docker icon and go into Preferences..., click into the Daemon tag.
+Add `10.10.0.2:5000` to Insecure registries.
+Finally click the `Apply and Start` button in the bottom to restart Docker with new setting.
+The final setup should be like this:
+![Docker Daemon on macOS](macos_docker_daemon.png)
+
+## 3. Build istio images
+Push images from your local dev environment to local registry on vagrant vm:
+```bash
+sh test_setup.sh
+```
+You should push new images whenever you modify istio source code.
+
+## 4. Run tests!
+### 1. Run tests on host machine
+You can issue test command on your host machine.
+E.g.
+```bash
+cd $ISTIO/istio
+make e2e_simple E2E_ARGS="--use_local_cluster" HUB=10.10.0.2:5000 TAG=latest
+```
+You can keep repeating this step if you made any local changes and want to run e2e tests again.
+Add E2E_ARGS="--use_local_cluster" to all your e2e tests as tests are we are running a local cluster.
```

Comment:
nit: as we are running tests on a local cluster?

Reply:
Done.